lib/collection/src/problems/unindexed_field.rs

```diff
@@ -533,11 +507,117 @@ impl<'a> Extractor<'a> {
+    let mut index_types = HashSet::new();
+    match value {
+        PayloadFieldSchema::FieldType(payload_schema_type) => match payload_schema_type {
+            PayloadSchemaType::Keyword => index_types.insert(FieldIndexType::KeywordMatch),
+            PayloadSchemaType::Integer => index_types.insert(FieldIndexType::IntMatch),
+            PayloadSchemaType::Uuid => index_types.insert(FieldIndexType::UuidMatch),
+            PayloadSchemaType::Bool => index_types.insert(FieldIndexType::BoolMatch),
+            PayloadSchemaType::Float => index_types.insert(FieldIndexType::FloatRange),
+            PayloadSchemaType::Geo => index_types.insert(FieldIndexType::Geo),
+            PayloadSchemaType::Text => index_types.insert(FieldIndexType::Text),
+            PayloadSchemaType::Datetime => index_types.insert(FieldIndexType::DatetimeRange),
+        },
+        PayloadFieldSchema::FieldParams(payload_schema_params) => match payload_schema_params {
+            PayloadSchemaParams::Keyword(_) => index_types.insert(FieldIndexType::KeywordMatch),
+            PayloadSchemaParams::Integer(integer_index_params) => {
+                if integer_index_params.lookup == Some(true) {
+                    index_types.insert(FieldIndexType::IntMatch);
+                }
+                if integer_index_params.range == Some(true) {
+                    index_types.insert(FieldIndexType::IntRange);
+                }
+                debug_assert!(
+                    !index_types.is_empty(),
+                    "lookup or range must be true for Integer payload index"
+                );
+                // unifying match arm types
+                true
+            }
+            PayloadSchemaParams::Uuid(_) => index_types.insert(FieldIndexType::UuidMatch),
+            PayloadSchemaParams::Bool(_) => index_types.insert(FieldIndexType::BoolMatch),
+            PayloadSchemaParams::Float(_) => index_types.insert(FieldIndexType::FloatRange),
+            PayloadSchemaParams::Geo(_) => index_types.insert(FieldIndexType::Geo),
+            PayloadSchemaParams::Text(_) => index_types.insert(FieldIndexType::Text),
+            PayloadSchemaParams::Datetime(_) => index_types.insert(FieldIndexType::DatetimeRange),
+        },
+    };
+    index_types
+}
+
+impl From<FieldIndexType> for PayloadFieldSchema {
+    fn from(val: FieldIndexType) -> Self {
+        match val {
+            FieldIndexType::IntMatch => {
+                let params = IntegerIndexParams {
+                    lookup: Some(true),
+                    ..Default::default()
+                };
+                PayloadFieldSchema::FieldParams(PayloadSchemaParams::Integer(params))
+            }
+            FieldIndexType::IntRange => {
+                let params = IntegerIndexParams {
+                    range: Some(true),
+                    ..Default::default()
+                };
+                PayloadFieldSchema::FieldParams(PayloadSchemaParams::Integer(params))
+            }
```

Comment:
I think we should treat the parametrized versions as advanced usage. Since we don't issue a recommendation for on_disk/is_tenant/etc settings, I'd say we can just convert to unparametrized type index here.

Moreover, this change would make it more obvious to convert to `PayloadSchemaType` instead.

Wdyt?

Reply:
Done in https://github.com/qdrant/qdrant/pull/6496/commits/8bc30336e22f5e26b93e3dd9e7250a058606719d :+1: